pkg/scheduler/factory/factory_test.go

```diff
@@ -264,7 +265,51 @@ func TestDefaultErrorFunc(t *testing.T) {
 		// whole error handling system in the future. The test will time
 		// out if something doesn't work.
 		time.Sleep(10 * time.Millisecond)
-		got, exists, _ := queue.Get(testPod)
+		got, exists, _ := queue.GetUnschedulableOrBackoffPod(testPod)
+		if !exists {
+			continue
+		}
+		requestReceived := false
+		actions := client.Actions()
+		for _, a := range actions {
+			if a.GetVerb() == "get" {
+				getAction, ok := a.(clienttesting.GetAction)
+				if !ok {
+					t.Errorf("Can't cast action object to GetAction interface")
+					break
+				}
+				name := getAction.GetName()
+				ns := a.GetNamespace()
+				if name != "foo" || ns != "bar" {
+					t.Errorf("Expected name %s namespace %s, got %s %s",
+						"foo", "bar", name, ns)
+				}
+				requestReceived = true
+			}
+		}
+		if !requestReceived {
+			t.Errorf("Get pod request not received")
+		}
+		if e, a := testPod, got; !reflect.DeepEqual(e, a) {
+			t.Errorf("Expected %v, got %v", e, a)
+		}
+		break
+	}
+
+	// Remove the pod from priority queue to test putting error
+	// pod in backoff queue.
+	queue.Delete(testPod)
+
+	// Trigger a move request
+	queue.MoveAllToActiveQueue()
+
+	// Trigger error handling again to put the pod in backoff queue
+	errFunc(testPod, nil)
+
+	for {
+		time.Sleep(10 * time.Millisecond)
+		// The pod should be found from backoff queue at this time
+		got, exists, _ := queue.GetUnschedulableOrBackoffPod(testPod)
```

Comment:
if pod doesn't exist, the test does not fail. It just goes to an infinite loop.

Reply:
True. Since the test is using a fake client to get pods, the pod should be accessible if there are no other bugs. I am thinking it should be safe to loop until waiting for up to a minute. If it is still timeout, test should be treated as failure.

Reply:
Yeah, or just simply try a maximum of 100 iterations of the for loop and if it still cannot find the pod, fail the test.

Reply:
Updated.